Description: Switch from lw_vasprintf() to sqlite3_vmprintf().
 lw_vasprintf() was removed in liblwgeom 2.2.0.
Author: Alessandro Furieri <a.furieri@lqt.it>
Origin: https://www.gaia-gis.it/fossil/libspatialite/fdiff?v1=48918773ced633bc&v2=342a605d691bddc6&sbs=1
        https://www.gaia-gis.it/fossil/libspatialite/fdiff?v1=bae06cd7cce4c2c9&v2=6000f2931ddaf06d&sbs=1
Bug-Debian: https://bugs.debian.org/807556

--- a/src/gaiageo/gg_lwgeom.c
+++ b/src/gaiageo/gg_lwgeom.c
@@ -90,29 +90,31 @@ splite_lwgeom_version (void)
 static void
 lwgaia_noticereporter (const char *fmt, va_list ap)
 {
-    char *msg;
-    if (!lw_vasprintf (&msg, fmt, ap))
+    char *msg = sqlite3_vmprintf (fmt, ap);
+    if (msg == NULL)
       {
 	  va_end (ap);
 	  return;
       }
-    spatialite_e ("LWGEOM notice: %s\n", msg);
+    if (strlen (msg) > 1)
+        spatialite_e ("LWGEOM notice: %s\n", msg);
     gaiaSetLwGeomWarningMsg (msg);
-    free (msg);
+    sqlite3_free (msg);
 }
 
 static void
 lwgaia_errorreporter (const char *fmt, va_list ap)
 {
-    char *msg;
-    if (!lw_vasprintf (&msg, fmt, ap))
+    char *msg = sqlite3_vmprintf (fmt, ap);
+    if (msg == NULL)
       {
 	  va_end (ap);
 	  return;
       }
-    spatialite_e ("LWGEOM error: %s\n", msg);
+    if (strlen (msg) > 1)
+        spatialite_e ("LWGEOM error: %s\n", msg);
     gaiaSetLwGeomErrorMsg (msg);
-    free (msg);
+    sqlite3_free (msg);
 }
 
 #ifndef POSTGIS_2_1
